src/lib/router.ts

```diff
@@ -30,4 +32,69 @@ router.post('/matches', async (req, res, next) => {
   }
 });
 
+type Highlight = {
+  timestamp: number;
+  type: string;
+  videoSrc: string;
+};
+
+router.post('/matches/:id/highlights', async (req, res, next) => {
+  const id = req.params.id;
+  const { timestamp, type, videoSrc } = req.body;
+
+  if (!isValid(id)) {
+    res.status(400).send('Invalid match id');
+    return;
+  }
+
+  if (
+    typeof timestamp !== 'number' ||
+    typeof type !== 'string' ||
+    typeof videoSrc !== 'string'
+  ) {
+    res.status(400).send('Invalid payload');
+    return;
+  }
+
+  const newHighlight: Highlight = {
+    timestamp,
+    type,
+    videoSrc,
+  };
+
+  try {
+    const updated = await getMatchesCollection().updateOne(
+      { _id: new ObjectId(id) },
+      { $push: { highlights: newHighlight } }
+    );
+
+    if (!updated.acknowledged || updated.modifiedCount === 0) {
+      res.status(404).send('match identifier incorrect');
+      return;
+    }
+
+    const result = await getMatchesCollection().findOne({
+      _id: new ObjectId(id),
+    });
```

Comment:
A small optimization would be the use of [ `findOneAndUpdate`](https://mongodb.github.io/node-mongodb-native/4.0/classes/collection.html#findoneandupdate) to reduce two requests (`updateOne` and `findOne`) to a single request (`findOneAndUpdate`).

Reply:
i wasn`t sure, that this works the same, but you are right, it would reduce server load